docs/source/spark.rst

```diff
@@ -18,84 +18,140 @@ has fewer features and, instead, is used in conjunction with other libraries,
+| :doc:`scalable multi-dimensional arrays <array>`.| arrays natively (this would be challenging given       |
+|                                                  | their computation model), although some support for    |
+|                                                  | two-dimensional matrices may be found in MLLib.        |
+|                                                  | People may also want to look at the                    |
+|                                                  | `Thunder <https://github.com/thunder-project/thunder>`_|
+|                                                  | project, which combines Apache Spark with NumPy arrays.|
++--------------------------------------------------+--------------------------------------------------------+
+
+Streaming API:
+
++--------------------------------------------------------+--------------------------------------------------------+
+| **Dask**                                               | **Spark**                                              |
++--------------------------------------------------------+--------------------------------------------------------+
+| Provides a :doc:`real-time futures interface <futures>`| Support for streaming data is first-class and          |
+| that is lower-level than Spark streaming. This enables | integrates well into their other APIs. It follows      |
+| more creative and complex use-cases, but requires      | a mini-batch approach.  This provides decent           |
+| more work than Spark streaming.                        | performance on large uniform streaming operations.     |
++--------------------------------------------------------+--------------------------------------------------------+
+
+Graphs / complex networks:
+
++--------------------------------------------------------+--------------------------------------------------------+
+| **Dask**                                               | **Spark**                                              |
++--------------------------------------------------------+--------------------------------------------------------+
+| Provides no such library.                              | Provides GraphX, a library for graph processing.       |
++--------------------------------------------------------+--------------------------------------------------------+
+
+Custom parallelism:
+
++--------------------------------------------------------+--------------------------------------------------------+
+| **Dask**                                               | **Spark**                                              |
++--------------------------------------------------------+--------------------------------------------------------+
+| Allows you to specify arbitrary task graphs for more   | Generally expects users to compose computations out    |
+| complex and custom systems that are not part of the    | of their high-level primitives (map, reduce, groupby,   |
```

Comment:
I'm getting `WARNING: Malformed table.` for the "Custom parallelism" table when building the docs locally. This should fix it:

```suggestion
| complex and custom systems that are not part of the    | of their high-level primitives (map, reduce, groupby,  |
```

Reply:
Aha, OK. It really shouldn't have to line up...